src/Altinn.App.Core/Infrastructure/Clients/Authorization/AuthorizationClient.cs

```diff
@@ -179,4 +180,52 @@ List<string> actions
         }
         return MultiDecisionHelper.ValidatePdpMultiDecision(actionsResult, response.Response, user);
     }
+
+    /// <summary>
+    /// Retrieves roles for a user on a specified party.
+    /// </summary>
+    /// <param name="userId">The user id.</param>
+    /// <param name="userPartyId">The user party id.</param>
+    /// <returns>A list of roles for the user on the specified party.</returns>
+public async Task<List<Role>> GetUserRolesAsync(int userId, int userPartyId)
+{
+    List<Role> roles = new();
+    string apiUrl = $"roles?coveredByUserId={userId}&offeredByPartyId={userPartyId}";
+    string token = JwtTokenUtil.GetTokenFromContext(_httpContextAccessor.HttpContext, _settings.RuntimeCookieName);
+
+    try
+    {
+        HttpResponseMessage response = await _client.GetAsync(token, apiUrl);
+        if (response.IsSuccessStatusCode)
+        {
+            string responseContent = await response.Content.ReadAsStringAsync();
+            var deserialized = JsonConvert.DeserializeObject<List<Role>>(responseContent);
+            if (deserialized is not null)
+            {
+                roles = deserialized;
+            }
+        }
+        else
+        {
+            _logger.LogError(
+                "Failed to retrieve roles for userId {UserId} and partyId {PartyId}. StatusCode: {StatusCode}",
+                userId,
+                userPartyId,
+                response.StatusCode
+            );
+        }
+    }
+    catch (Exception ex)
+    {
+        _logger.LogError(
+            ex,
+            "An error occurred while retrieving roles for userId {UserId} and partyId {PartyId}",
+            userId,
+            userPartyId
+        );
```

Comment:
I'm not sure about the error handling here. If the API fails for some reason we return an empty list, which might make it look like the user has no roles when in fact the problem was that we couldn't find out at all

Reply:
Good point.